src/app/modules/submission/encrypt-submission/encrypt-submission.service.ts

```diff
@@ -256,6 +288,46 @@ export const getEncryptedSubmissionData = (
   })
 }
 
+/**
+ * Gets completed payment details associated with a particular submission for a
+ * given paymentId.
+ * @param paymentId the payment
+ * @requires paymentId must be a completed payment
+ *
+ * @returns ok(SubmissionPayment)
+ * @returns err(PaymentNotFoundError) if the paymentId does not reference a payment, or if the payment is incomplete
+ * @returns err(DatabaseError) if mongoose threw an error during the process
+ */
+export const getSubmissionPaymentDto = (
+  paymentId: string,
+): ResultAsync<SubmissionPaymentDto, PaymentNotFoundError | DatabaseError> =>
+  PaymentsService.findPaymentById(paymentId).andThen((payment) => {
+    // If the payment is incomplete, the "complete payment" is not found. This
+    // also implies an internal consistency error.
+    if (!payment.completedPayment) return errAsync(new PaymentNotFoundError())
+
+    return okAsync({
+      id: payment._id,
+      paymentIntentId: payment.paymentIntentId,
+      email: payment.email,
+      amount: payment.amount,
+      status: payment.status,
+
+      paymentDate: moment(payment.completedPayment.paymentDate)
```

Comment:
Unrelated to your PR but I realised that our BE uses [moment](https://momentjs.com/docs/#/-project-status/) which is now legacy? 

Whilst FE (and BE tests) mostly use date-fns, was wondering if we should just start using date-fns in general?

Reply:
yup, maybe will need to migrate again sometime in the future, probably better to do all at once